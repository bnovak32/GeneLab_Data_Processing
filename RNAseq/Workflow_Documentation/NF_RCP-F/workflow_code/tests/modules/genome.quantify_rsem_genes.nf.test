nextflow_process {

    name "Test Process QUANTIFY_RSEM_GENES"
    script "modules/genome.nf"
    process "QUANTIFY_RSEM_GENES"

    test("Baseline_ON_GLDS194:Should run without failures") {

        when {
            params {
                // define parameters here. Example:
                // outdir = "tests/results"
            }
            process {
                """
                // define inputs of the process here. Example:
                input[0] = file("test-datasets-extended/testdata/GLDS-194/mock/samples.txt")
                input[1] = file("test-datasets-extended/testdata/GLDS-194/03-RSEM_Counts/*.genes.results")
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out.publishables).match()
        }

    }

    test("Issue_60_Test_ON_Mock:Should run without failures") {

        when {
            params {
                // define parameters here. Example:
                // outdir = "tests/results"
            }
            process {
                """
                // define inputs of the process here. Example:
                input[0] = file("test-datasets-extended/testdata/mocks/overlapping_samplenames/samples.txt")
                input[1] = file("test-datasets-extended/testdata/mocks/overlapping_samplenames/*.genes.results")
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out.publishables).match()
        }

    }

}
