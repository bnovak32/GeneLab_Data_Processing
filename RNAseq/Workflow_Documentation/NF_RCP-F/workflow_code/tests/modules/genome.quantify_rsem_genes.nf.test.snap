{
    "Baseline_ON_GLDS194:Should run without failures": {
        "content": [
            [
                [
                    "RSEM_Unnormalized_Counts.csv:md5,f2176089284f6018530dbb8b6618b0aa",
                    "RSEM_NumNonZeroGenes.csv:md5,6dd712284c01af420ed6a31a0256e516"
                ]
            ]
        ],
        "timestamp": "2023-01-25T20:37:40+0000"
    },
    "Issue_60_Test_ON_Mock:Should run without failures": {
        "content": [
            [
                [
                    "RSEM_Unnormalized_Counts.csv:md5,9af39c67a784e695ec553606b44d0115",
                    "RSEM_NumNonZeroGenes.csv:md5,699ee0a7846a173d93f02d27c28caedd"
                ]
            ]
        ],
        "timestamp": "2023-01-25T20:37:40+0000"
    }
}